app/code/Magento/Catalog/etc/config.xml

```diff
@@ -41,7 +41,7 @@
             <seo>
                 <product_url_suffix>.html</product_url_suffix>
                 <category_url_suffix>.html</category_url_suffix>
-                <product_use_categories>0</product_use_categories>
+                <product_use_categories>1</product_use_categories>
```

Comment:
Hey @vshatylo This is not a fix of the test. You should modify failing tests by adding there setting of the desired value for this setting (in phpDoc annotations) instead of changing the default behavior of Magento.

Reply:
Hi @slavvka just confirm that need to change test? 
Based on changes in  app/code/Magento/Catalog/Model/Product/Url.php
before changes default behavior expects that category path available in product url in case it visited from category overview page.
![screenshot from 2019-02-01 08-38-50](https://user-images.githubusercontent.com/43958431/52106919-03aaf580-25fd-11e9-8710-fa7f590fa6b7.png)

Reply:
Yes, I took a look at one of the faili tests and saw that it just doesn't account that settings. You need to update tests to ensure that their preconditions are valid.